Models/getArticleById.model.js

```diff
@@ -0,0 +1,12 @@
+const db = require("../db/connection");
+
+exports.getArticleById = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        throw new Error(`Article with ID ${article_id} not found.`);
```

Comment:
A promise.reject would work better here so that we can also send the status code we want for this error. returning a promise.reject will send the error object we pass into it to the .catch() inside the controller which can be passed to the error handling middleware using next()